kakarot_scripts/utils/kakarot.py

```diff
@@ -747,29 +745,97 @@ async def store_bytecode(bytecode: Union[str, bytes], **kwargs):
     return evm_address
 
 
-async def deploy_with_presigned_tx(
-    deployer_evm_address: str, signed_tx: bytes, amount=0.1, name="", max_fee=None
-):
-    deployer_starknet_address = await deploy_and_fund_evm_address(
-        deployer_evm_address, amount
+# async def deploy_with_presigned_tx(
+#     deployer_evm_address: str, signed_tx: bytes, amount=0.1, name="", max_fee=None
+# ):
+#     deployer_starknet_address = await deploy_and_fund_evm_address(
+#         deployer_evm_address, amount
+#     )
+#     nonce = (
+#         await _call_starknet(
+#             "account_contract", "get_nonce", address=deployer_starknet_address
+#         )
+#     ).nonce
+#     if nonce != 0:
+#         logger.info(
+#             f"ℹ️  Nonce for {deployer_evm_address} is not 0 ({nonce}), skipping transaction"
+#         )
+#         return
+
+#     response = await send_pre_eip155_transaction(
+#         deployer_evm_address, deployer_starknet_address, signed_tx, max_fee
+#     )
+
+#     receipt, response, success, gas_used = response
+#     deployed_address = response[1]
+#     logger.info(f"✅ {name} Deployed at: 0x{deployed_address:040x}")
+#     deployed_starknet_address = await get_starknet_address(deployed_address)
+#     return {"address": deployed_address, "starknet_address": deployed_starknet_address}
+
+
+async def whitelist_pre_eip155_tx(name: str) -> bool:
```

Comment:
so the whitelist function should
- analyze the tx to get the corresponding nonce
- call kakarot to get the deployer nonce
- skip if tx nonce < kakarot nonce
- invoke authorized otherwise

no eoa deployment here